test/Job.test.js

```diff
@@ -136,215 +134,182 @@ describe('Testing Job class', function() {
+        let jobId;
+        return client
+          .service(testTasks.sleepTask.service)
+          .task(testTasks.sleepTask.name)
+          .submit({inputParameters: testTasks.sleepTaskFail.parameters})
           .then((job) => {
-            job.once('Succeeded', function(data) {
-              succeededEventCalled = true;
-            });
-            job.once('Failed', function(data) {
-              try {
-                expect(data).to.exist;
-                expect(data).to.be.an('object');
-                expect(succeededEventCalled).to.be.false;
-                done();
-              } catch (e) {
-                done(e);
-              }
-
-            });
-          }).catch((err) => {
-            done('Error submitting job: ' + err);
+            job.on('Succeeded', succeededListener);
+            job.on('Failed', failedListener);
+            jobId = job.jobId;
+            return job.wait();
+          })
+          .catch(() => {
+            return Promise.all([
+              assert(failedListener.calledOnceWith({jobId: jobId, success: false})),
+              assert(succeededListener.notCalled)
+            ]);
           });
       });
     });
 
     describe('\'Progress\' event', function() {
-      it('fires when job emits progress', function(done) {
+      it('fires when job emits progress', function() {
         this.timeout(config.testTimeout2);
 
-        let nProgressEvents = 0;
-        let testData = testTasks;
+        let jobId = null;
+        let testData = Object.assign({}, testTasks);
         const nProgress = 5;
         const progressMessage = 'Message';
         testData.sleepTask.parameters.N_PROGRESS = nProgress;
         testData.sleepTask.parameters.PROGRESS_MESSAGE = progressMessage;
 
-        const task = new Task(server.service(testTasks.sleepTask.service),
-          testTasks.sleepTask.name);
-        task.submit({parameters: testTasks.sleepTask.parameters})
-          .then((job2) => {
-            job2.on('Progress', function(data) {
-              try {
-                nProgressEvents++;
-                expect(data.progress).to.exist;
-                expect(data.progress).to.be.a('number');
-                expect(data.progress).to.be.above(-1);
-                expect(data.progress).to.be.below(100);
-                expect(data.message).to.exist;
-                expect(data.message).to.equal(progressMessage);
-              } catch (e) {
-                done(e);
-              }
-
-            });
-            job2.once('Succeeded', function(data) {
-              try {
-                expect(nProgressEvents).to.equal(nProgress);
-                done();
-              } catch (e) {
-                done(e);
-              }
-
-            });
-            job2.once('Failed', function(data) {
-              done('Failed');
+        const progressListener = sinon.spy();
+        const completedListener = sinon.spy();
+
+        setTimeout(() => {
+          return client
```

Comment:
can you return a promise from inside a timeout?

Reply:
I think I added that while debugging - we don't need it.  Taking it out.